docs/guides/building-forms.md

```diff
@@ -0,0 +1,387 @@
+
+![form-no-js](https://assets.vercel.com/image/upload/c_scale,w_675/v1643009087/nextjs/guides/building-forms/form-no-js.jpg)
+
+## Part 3: Forms in Next.js
+
+You have created a server with Vercel via Serverless Functions. Now it's time to configure the client (the form itself) inside Next.js using React. The first step will be extending your knowledge of HTML forms and converting it into React (using [JSX](https://reactjs.org/docs/introducing-jsx.html)).
+
+Here's the same form in a [React function component](https://reactjs.org/docs/components-and-props.html) written using [JSX](https://reactjs.org/docs/introducing-jsx.html).
+
+```js
+export default function Form() {
+  return (
+    <form action="/api/form" method="post">
+      <label htmlFor="first">First Name</label>
+      <input type="text" id="first" name="first" required />
+
+      <label htmlFor="last">Last Name</label>
+      <input type="text" id="last" name="last" required />
+
+      <button type="submit">Submit</button>
+    </form>
+  )
+}
+```
+
+Here's what changed:
+
+- The `for` attribute is changed to `htmlFor`. (Since `for` is a keyword associated with the "for" loop in JavaScript, React elements use `htmlFor` instead.)
+- The `action` attribute now has a relative URL, the form API endpoint deployed at Vercel.
+
+This completes the basic structure of your Next.js-based form.
+
+> You can view the entire source code of [next-forms](https://github.com/vercel/next.js/tree/canary/examples/next-forms) example repo that we're creating here as a working example. Feel free to clone it and start right away. This demo is built with create-next-app, and you can preview the basic form CSS styles inside `/styles/global.css` file.
+
+![forms with nextjs](https://assets.vercel.com/image/upload/c_scale,w_675/v1643009088/nextjs/guides/building-forms/forms-with-nextjs.png)
+
+## Part 4: Form Submission without JavaScript
+
+JavaScript brings interactivity to our web applications, but sometimes you need to control the JavaScript bundle from being too large, or your sites visitors might have JavaScript disabled.
+
+There are several reasons why users disable JavaScript, e.g., addressing bandwidth constraints, increasing device (phone or laptop) battery life, and even for privacy so they won’t be tracked with analytical scripts. Regardless of the reason, disabling JavaScript will impact site functionality partially, if not completely.
+
+But with Vercel serverless functions, we can still use the forms without JavaScript, as we saw in the **Form Submission without JavaScript** [section](https://github.com/MaedahBatool/next.js/blob/forms/docs/guides/building-forms.md#form-submission-without-javascript) above.
```

Comment:
Here we are linking to the current section this paragraph is in. Should it link somewhere else?

```suggestion
But with Vercel serverless functions, we can still use the forms without JavaScript, as seen in the **Form Submission without JavaScript** [section](#form-submission-without-javascript) above.
```